DeviceSimulation/SimulationTests.cs

```diff
@@ -1,33 +1,302 @@
+            Task<CloudToDeviceMethodResult> rebootMethodResult = serviceClient.InvokeDeviceMethodAsync("chiller-01.0",
+                new CloudToDeviceMethod("Reboot"));
+            CloudToDeviceMethodResult rebootResponse = await rebootMethodResult.ConfigureAwait(false);
+
+            //Assert
+            Assert.Equal(200, rebootResponse.Status);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Start_Given_Simulation()
+        {
+            //Arrange
+            string ETag = this.get_ETag_Of_Running_Simulation();
+
+            var simulationContent = "{" + $"'ETag': '{ETag}' ,'Enabled': true" + "}";
+            var simulationContentByteArray = Encoding.ASCII.GetBytes(simulationContent);
+
+            //Act
+            HttpWebRequest startSimulationRequest = this.Create_Simulation_Request(simulationContentByteArray);
+            Stream dataStream = startSimulationRequest.GetRequestStream();
+            dataStream.Write(simulationContentByteArray, 0, simulationContentByteArray.Length);
+            dataStream.Close();
+            var startSimulationResponse = (HttpWebResponse)startSimulationRequest.GetResponse();
+
+            //Assert
+            Assert.Equal(HttpStatusCode.OK, startSimulationResponse.StatusCode);
+
+            var verificationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var verificationResponse = this.httpClient.GetAsync(verificationRequest).Result;
+            JObject jsonResponse = JObject.Parse(verificationResponse.Content);
+
+            Assert.True((bool)jsonResponse["Enabled"]);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Stop_Given_Simulation()
+        {
+            //Arrange
+            string ETag = this.get_ETag_Of_Running_Simulation();
+
+            var simulationContent = "{" + $"'ETag': '{ETag}' ,'Enabled': false" + "}";
+            var simulationContentByteArray = Encoding.ASCII.GetBytes(simulationContent);
+
+            //Act
+            HttpWebRequest stopSimulationRequest = this.Create_Simulation_Request(simulationContentByteArray);
+            Stream dataStream = stopSimulationRequest.GetRequestStream();
+            dataStream.Write(simulationContentByteArray, 0, simulationContentByteArray.Length);
+            dataStream.Close();
+            var stopSimulationResponse = (HttpWebResponse)stopSimulationRequest.GetResponse();
+
+            //Assert
+            Assert.Equal(HttpStatusCode.OK, stopSimulationResponse.StatusCode);
+
+            var verificationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var verificationResponse = this.httpClient.GetAsync(verificationRequest).Result;
+            JObject jsonResponse = JObject.Parse(verificationResponse.Content);
+
+            Assert.False((bool)jsonResponse["Enabled"]);
+        }
+
+        private string get_ETag_Of_Running_Simulation()
```

Comment:
this method is deleting existing simulation, re-creating a new one and returning etag, should just return etag of existing simulation

Reply:
This is to handle the case where the simulation with id 1 has been deleted. If the simulation with id 1 has been deleted due to some course of action, then there would be no simulation and getting the ETag would fail. 

Please let me know if I am wrong.